test/runtime/chaos.go

```diff
@@ -127,4 +133,58 @@ var _ = Describe("RuntimeChaos", func() {
 		Expect(fds).To(BeNumerically("<", threshold),
 			fmt.Sprintf("%d file descriptors open from Cilium processes", fds))
 	}, 300)
+
+	It("Checking that during restart no traffic is dropped using Egress + Ingress Traffic", func() {
+		By("Installing sample containers")
+		vm.SampleContainersActions(helpers.Create, helpers.CiliumDockerNetwork)
+		vm.PolicyDelAll().ExpectSuccess("Cannot deleted all policies")
+
+		_, err := vm.PolicyImportAndWait(vm.GetFullPath(policiesL4Json), helpers.HelperTimeout)
+		Expect(err).Should(BeNil(), "Cannot install L4 policy")
+
+		areEndpointsReady := vm.WaitEndpointsReady()
+		Expect(areEndpointsReady).Should(BeTrue(), "Endpoints are not ready after timeout")
+
+		By("Starting background connection from app2 to httpd1 container")
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		srvIP, err := vm.ContainerInspectNet(helpers.Httpd1)
+		Expect(err).Should(BeNil(), "Cannot get httpd1 server address")
+		type BackgroundTestAsserts struct {
+			res  *helpers.CmdRes
+			time time.Time
+		}
+		backgroundChecks := []*BackgroundTestAsserts{}
+		go func() {
+			for {
+				select {
+				default:
+					res := vm.ContainerExec(
+						helpers.App1,
+						helpers.CurlFail("http://%s/", srvIP[helpers.IPv4]))
+					assert := &BackgroundTestAsserts{
+						res:  res,
+						time: time.Now(),
+					}
+					backgroundChecks = append(backgroundChecks, assert)
+				case <-ctx.Done():
+					return
+				}
+			}
+		}()
+		// Sleep a bit to make sure that the goroutine starts.
+		time.Sleep(50 * time.Millisecond)
+
+		err = vm.RestartCilium()
+		Expect(err).Should(BeNil(), "restarting Cilium failed")
+
+		By("Stopping background connections")
+		cancel()
```

Comment:
This will not do what you will be expecting, when calling `cancel()` it doesn't mean that the go routine created in line 158 is completed. Add a sync.WaitGroup in line 171 with `wg.Done()` and do `wg.Wait()` here.